### PR TITLE
Vtk opengl2 build

### PR DESCRIPTION
**Pull request by danlipsa**

### CODE REVIEW

File: vtk-cdat/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+ffmpeg:
+    - 4.0.*
+pin_run_as_build:
+  ffmpeg: x.x
+
+numpy:
+    - 1.15
+
+VTK_WITH_OSMESA:
```
Comment (doutriaux1) [inline vtk-cdat/conda_build_config.yaml]: I think you took off the "feature" capability, no? I use to determine this from a "conda list"

File: vtk-cdat/meta.yaml.in [hunk 2]
```diff
@@ -1,44 +1,35 @@
...
   build:
+    - toolchain
     - cmake
-    - python
-    - proj4 <5
-    - hdf5 1.10.*
-    - ffmpeg
-    - freetype
-    - mesalib >17  # {{{ mesalib }}}
-    - x264
```
Comment (doutriaux1) [inline vtk-cdat/meta.yaml.in]: we want this (x264) otherwise it's bad videos.

File: vtk-cdat/conda_build_config.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+python:
+    - 3.6
+    - 2.7
+pin_run_as_build:
+  python: x.x
+
+ffmpeg:
+    - 4.0.*
+pin_run_as_build:
```
Comment (doutriaux1) [inline vtk-cdat/conda_build_config.yaml]: nice one.

**Comment by danlipsa** [comment]
@scottwitenburg Please review

**Comment by doutriaux1** [comment]
@danlipsa trying on my mac.

File: vtk-cdat/conda_build_config.yaml (hunk 1 above)
Comment (danlipsa) [reply vtk-cdat/conda_build_config.yaml]: I think you generated two different meta.yaml from prep_from_build. The advantage of this is that it builds all 4 configurations with one conda build.

**Comment by doutriaux1** [comment]
@danlipsa I used to run `prep_for_build.py` in the root. This uses `meta.yaml.in` so runnnig this will overwrite your changes. This script allows to decide on the branch to use, version name, and mesalib on/off, among other things. It's ok for now but we will need to move your meta.yaml to meata.yaml.in at some point.

File: vtk-cdat/meta.yaml.in (hunk 2 above)
Comment (danlipsa) [reply vtk-cdat/meta.yaml.in]: Sounds good. I can add that - I wasn't sure what was that for.

File: vtk-cdat/conda_build_config.yaml (hunk 3 above)
Comment (danlipsa) [reply vtk-cdat/conda_build_config.yaml]: Yes. I just tried it and it works! Initially Initially I tried it without pinning a version for ffmpeg (just left conda to pick up the latest version) but I could not make that work.

**Comment by doutriaux1** [comment]
my bad, you did change the .in file. Thanks!

**Comment by danlipsa** [comment]
I used this this recipe to build all 4 configurations on linux (about 40 min on my desktop) and on mac (about 3 hours on my laptop) using just one conda build on each platform. VTK still needs a couple of commits from Scott but seems that libraries match.

**Comment by doutriaux1** [comment]
@danlipsa on my mac:
```
doutriaux1@loki:[conda-recipes]:[vtk-opengl2-build]:[45494]> conda build vtk-cdat -c conda-forge
Cloning into bare repository '/Users/doutriaux1/anaconda2/conda-bld/git_cache/gitlab.kitware.com/vtk/vtk.git'...
remote: Enumerating objects: 546748, done.
remote: Counting objects: 100% (546748/546748), done.
remote: Compressing objects: 100% (124491/124491), done.
remote: Total 546748 (delta 417530), reused 537701 (delta 413073)
Receiving objects: 100% (546748/546748), 204.38 MiB | 2.36 MiB/s, done.
Resolving deltas: 100% (417530/417530), done.
Cloning into '/Users/doutriaux1/anaconda2/conda-bld/vtk-cdat_1538418969116/work'...
done.
Checking out files: 100% (16453/16453), done.
checkout: 'master'
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Submodule 'VTK-m' (https://gitlab.kitware.com/vtk/vtk-m.git) registered for path 'ThirdParty/vtkm/vtk-m'
Cloning into '/Users/doutriaux1/anaconda2.new/conda-bld/vtk-cdat_1538418969116/work/ThirdParty/vtkm/vtk-m'...
git-lfs filter-process: git-lfs: command not found
fatal: The remote end hung up unexpectedly
Unable to checkout '8d0f441e1b391c6ec4c2662ee1454d8c901733fb' in submodule path 'ThirdParty/vtkm/vtk-m'
Warning: failed to download source.  If building, will try again after downloading recipe dependencies.
Error was: 
Command '['/usr/bin/git', 'submodule', 'update', '--init', '--recursive']' returned non-zero exit status 1
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /git/conda-recipes/vtk-cdat/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /git/conda-recipes/vtk-cdat/conda_build_config.yaml
Cloning into '/Users/doutriaux1/anaconda2/conda-bld/vtk-cdat_1538418969116/work'...
done.
Checking out files: 100% (16453/16453), done.
checkout: 'master'
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Submodule 'VTK-m' (https://gitlab.kitware.com/vtk/vtk-m.git) registered for path 'ThirdParty/vtkm/vtk-m'
Cloning into '/Users/doutriaux1/anaconda2.new/conda-bld/vtk-cdat_1538418969116/work/ThirdParty/vtkm/vtk-m'...
git-lfs filter-process: git-lfs: command not found
fatal: The remote end hung up unexpectedly
Unable to checkout '8d0f441e1b391c6ec4c2662ee1454d8c901733fb' in submodule path 'ThirdParty/vtkm/vtk-m'
Warning: failed to download source.  If building, will try again after downloading recipe dependencies.
Error was: 
Command '['/usr/bin/git', 'submodule', 'update', '--init', '--recursive']' returned non-zero exit status 1
Attempting to finalize metadata for vtk-cdat
INFO:conda_build.metadata:Attempting to finalize metadata for vtk-cdat
Undefined Jinja2 variables remain (['GIT_DESCRIBE_TAG', 'GIT_DESCRIBE_TAG', 'GIT_DESCRIBE_NUMBER', 'GIT_DESCRIBE_HASH']).  Please enable source downloading and try again.
```

**Comment by doutriaux1** [comment]
i think it was gitlfs related

**Comment by doutriaux1** [comment]
how do you get gitlfs enabled?

**Comment by danlipsa** [comment]
Indeed, I don't have gitlfs on my machine. Did you started using that in cdat?

**Comment by danlipsa** [comment]
Maybe gitlfs was not tagged? The repo needs to be tagged to generate the version, build number and SHA.

**Comment by doutriaux1** [comment]
No I dont. I'm not sure why it's picking this up. Are you saying it's not needed on your mac?

**Comment by doutriaux1** [comment]
remove it from my ${HOME}/.gitconfig and it seems to work now.

**Comment by doutriaux1** [comment]
next error
```
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /git/conda-recipes/vtk-cdat/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /git/conda-recipes/vtk-cdat/conda_build_config.yaml
Attempting to finalize metadata for vtk-cdat
INFO:conda_build.metadata:Attempting to finalize metadata for vtk-cdat
Solving environment: ...working... done
Solving environment: ...working... done
Traceback (most recent call last):
  File "/Users/doutriaux1/anaconda2/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 420, in main
    execute(sys.argv[1:])
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/cli/main_build.py", line 411, in execute
    verify=args.verify)
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/api.py", line 200, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/build.py", line 2184, in build_tree
    notest=notest,
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/build.py", line 1197, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 670, in expand_outputs
    for (output_dict, m) in _m.get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/metadata.py", line 1972, in get_output_metadata_set
    bypass_env_check=bypass_env_check)
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/metadata.py", line 699, in finalize_outputs_pass
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 492, in finalize_metadata
    for dep in run_deps]
  File "/Users/doutriaux1/anaconda2/lib/python2.7/site-packages/conda_build/render.py", line 161, in get_pin_from_build
    pin = utils.apply_pin_expressions(version.split()[0], **pin_cfg)
TypeError: apply_pin_expressions() argument after ** must be a mapping, not unicode
```

**Comment by danlipsa** [comment]
Is this before anything starts building?

I am running this from anaconda3. Here is the versions I have:brasov:cdat-conda-recipes danlipsa$ which python
/Users/danlipsa/anaconda3/bin/python
brasov:cdat-conda-recipes danlipsa$ conda --version
conda 4.5.11
brasov:cdat-conda-recipes danlipsa$ conda build --version
conda-build 3.13.0
brasov:cdat-conda-recipes danlipsa$ 

Maybe you have some older conda / conda build?

**Comment by danlipsa** [comment]
You are using python 2. Try switching python 3 / conda 3.

**Comment by doutriaux1** [comment]
I have conda-build 3.11, trying to update

**Comment by danlipsa** [comment]
I wonder if its not python 2 version python 3. I build all 4 packages from python 3.

**Pull request merged**